app/components/elements/forms/select_day_field_component.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Elements
+  module Forms
+    # Component for form day of the month fields
+    class SelectDayFieldComponent < Forms::FieldComponent
+      def value
+        form.object.public_send(field_name)
```

Comment:
Does the Rails form magic not do this automagically?

Reply:
No, `select_day` isn't hooked into the form. See https://guides.rubyonrails.org/form_helpers.html#select-boxes-for-time-or-date-components

Reply:
Form-related "bare methods," TIL 🍻